warehouse/packaging/models.py

```diff
@@ -384,6 +384,17 @@ def github_repo_info_url(self):
                 user_name, repo_name = segments[:2]
                 return f"https://api.github.com/repos/{user_name}/{repo_name}"
 
+    @property
+    def gitlab_repo_info_url(self):
+        for parsed in [urlparse(url) for url in self.urls.values()]:
+            segments = parsed.path.strip("/").rstrip("/").split("/")
+
+            if (
+                parsed.netloc == "gitlab.com" or parsed.netloc == "www.gitlab.com"
+            ) and len(segments) >= 2:
+                user_name, repo_name = segments[:2]
+                return f"https://gitlab.com/api/v4/projects/{user_name}%2F{repo_name}"
```

Comment:
this property has a lot of common code with the above. It's not a blocker for the PR, but it'd be nice if we could refactor the common logic into a helper function.